File: rps-ts-server/src/domain/Game.ts
import { Move } from "./Move";
import { Result } from "./Result";

export function determineWinner(player1Move: Move, player2Move: Move): Result {
    if (player1Move == player2Move) {
        return Result.Draw;
    }

    if (
        (player1Move === Move.Rock && player2Move === Move.Scissors) ||
        (player1Move === Move.Paper && player2Move === Move.Rock) ||
        (player1Move === Move.Scissors && player2Move === Move.Scissors)
    ) {
        return Result.Player1Wins;
    }

    return Result.Player2Wins;
}

File: rps-rust-server/src/domain/game.rs
use super::game_move::Move;
use super::game_result::Result;

pub fn determine_winner(player1_move: &Move, player2_move: &Move) -> Result {
    use self::Move::*;
    use self::Result::*;

    match (player1_move, player2_move) {
        (Rock, Scissors) | (Paper, Rock) | (Scissors, Paper) => Player1Wins,
        (Scissors, Rock) | (Rock, Paper) | (Paper, Scissors) => Player2Wins,
        _ => Draw,
    }
}
